org.audiveris.audiveris.yml

```diff
@@ -0,0 +1,982 @@
+
+app-id: org.audiveris.audiveris
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+build-options:
+  append-path: "/usr/lib/sdk/openjdk17/bin"
+command: Audiveris
+finish-args:
+  # Network access
+  - --share=network
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=x11
+  # Wayland access
+  - --socket=wayland
+  # Needs to save files locally
+  - --filesystem=xdg-documents
+  - --filesystem=xdg-pictures
+  - --env=PATH=/app/bin:/app/jre/bin:/usr/bin
+  - --env=JAVA_HOME=/app/jre
+  # Calling abort() on malloc-check doesn't fly with OpenJDK
+  - --env=MALLOC_CHECK_=0
+modules:
+  - name: openjdk
+    buildsystem: simple
+    build-commands:
+      - /usr/lib/sdk/openjdk17/install.sh
+
+  - name: audiveris
+    buildsystem: simple
+    build-commands:
+      - ln -f gradle-7.3-all.zip gradle/wrapper/gradle-7.3-all.zip
+      - sed -Ei 's,https\\://services.gradle.org/distributions/,,' gradle/wrapper/gradle-wrapper.properties
+      - sh ./mkgradlerepo.sh
+      - install -m 0644 -D -t /app/share/tessdata *.traineddata
+      - install -m 0644 -D res/icon-256.png /app/share/icons/hicolor/256x256/apps/org.audiveris.audiveris.png
+      - install -m 0644 -D res/icon-64.png /app/share/icons/hicolor/64x64/apps/org.audiveris.audiveris.png
+      - install -m 0644 -D res/icon-50.png /app/share/icons/hicolor/50x50/apps/org.audiveris.audiveris.png
+      - install -m 0644 -D -m 644 org.audiveris.audiveris.desktop /app/share/applications/org.audiveris.audiveris.desktop
+      - install -m 0644 -D -m 644 org.audiveris.audiveris.metainfo.xml /app/share/metainfo/org.audiveris.audiveris.metainfo.xml
+      - ./gradlew --offline build
+      - tar x -C /app --strip-components=1 -f build/distributions/Audiveris-*.tar
+      - sed -i -f add-tessdata-prefix.sed /app/bin/Audiveris
+    sources:
+      - type: git
+        url: https://github.com/Audiveris/audiveris
+        branch: 5.3.1
+      - type: file
+        url: https://services.gradle.org/distributions/gradle-7.3-all.zip
+        sha256: 00b273629df4ce46e68df232161d5a7c4e495b9a029ce6e0420f071e21316867
+      - type: patch
+        path: 0001-add-local-dependencies-mirror.patch
+      - type: patch
+        path: 0001-WebBrowser-try-xdg-open-if-desktop.browse-fails.patch
+      - type: file
+        path: org.audiveris.audiveris.desktop
+      - type: file
+        path: org.audiveris.audiveris.metainfo.xml
+      - type: file
+        path: add-tessdata-prefix.sed
+      - type: file
+        # generated by deps.py
```

Comment:
then this should be a separate json to make it easier to regenerate. also where is a `deps.py`?

Reply:
I followed a different approach. The files in this submission are generated from templates in a separate project, https://github.com/mwilck/Audiveris-flatpak. The process involves building Audiveris locally, with network access, so that **gradle** can pull in all the Java dependencies. The `deps.py` script (part of the Audiveris-flatpak project) then collects the `.jar` files and respective meta data that **gradle** accumulated, and creates a list of `.jar` URLs with their respective checksums for flatpak to download, as well as the `mkgradlerepo.sh` script which will re-organize the downloaded libraries in a way that a network-less **gradle** build running inside `flatpak build` can understand.

Regenerating `org.audiveris.audiveris.yml` from the Audiveris-flatpak repository is quite straightforward.

I could have submitted to flathub from the Audiveris-flatpak project directly, but my thinking was that all this extra logic would be confusing, and ultimately doesn't belong here. Therefore I created a script that exports only the auto-generated data. The result of this script is what you see here.

If you think that I should rather submit from the Audiveris-flatpak repo directly, or have some other advice, please let me know.

I suppose I could also change the auto-generation process to use a "separate json" rather than regenerating the entire file, but if possible I would prefer doing that in a later, separate step. Can you point me to an example and/or documentation how to to provide the list of sources in a separate file? A quick web search didn't turn up anything for me.

Reply:
The comment about making the auto-generated dependencies a separate file stands. This is how every other auto-generated dependencies is used.

Reply:
Alright, I finally found the paragraph in the docs that explains how to do this in flatpak-manifest(5). I will work on adapting my scripts.